### PR TITLE
Add `Types of Activation Functions` file

### PR DESCRIPTION
**Pull request by Haleshot**
## 📝 Summary
Added an interactive notebook demonstrating various neural network activation functions using marimo. The notebook provides visual and mathematical explanations of 12 different activation functions commonly used in deep learning.

## 🔍 Description of Changes
- Created a new notebook `activation_functions.py` that provides:
  - Interactive visualization of activation functions using matplotlib
  - Mathematical formulas rendered with LaTeX
  - Dropdown menu to switch between different activation functions
  - Custom header widget with documentation links and metadata
  - Caching implementation for efficient plot generation
  
Key features:
- Covers 12 activation functions: Sigmoid, Tanh, ReLU, ELU, PReLU, Leaky ReLU, SELU, Softplus, Softsign, Hard Sigmoid, Swish, and Mish
- Real-time plotting with marimo's caching system
- Comprehensive documentation links to Python, Matplotlib, NumPy, and marimo docs
- Responsive design with show/hide functionality for long descriptions
- Mathematical formulas displayed using LaTeX

Dependencies:
- marimo
- anywidget==0.9.13
- traitlets==5.14.3
- numpy==2.1.2
- matplotlib==3.9.2

## 📋 Checklist
- [x] I have read the [contributor guidelines](../CONTRIBUTING.md)
- [x] I have run the code and verified that it works as expected
- [x] The notebook is properly documented with descriptions and references
- [x] All dependencies are specified in the requirements section
- [x] Code follows marimo's styling and best practices

## 📜 Reviewers
@metaboulie

### CODE REVIEW

File: marimo-tutorials/artificial-intelligence/deep-learning/activation-functions.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+            formula = r"$$\sigma(x) = \frac{1}{1 + e^{-x}}$$"
+        elif func_name == "tanh":
+            y = np.tanh(x)
+            formula = r"$$tanh(x) = \frac{e^x - e^{-x}}{e^x + e^{-x}}$$"
+        elif func_name == "relu":
+            y = np.maximum(0, x)
+            formula = r"$$ReLU(x) = max(0, x)$$"
+        elif func_name == "elu":
+            alpha = 1.0  # Default alpha value for ELU
+            y = np.where(x > 0, x, alpha * (np.exp(x) - 1))
```
Comment (metaboulie) [inline marimo-tutorials/artificial-intelligence/deep-learning/activation-functions.py:64]: For activation functions with hyper-parameters, we can add another slider to let reader control values

File: marimo-tutorials/Data-Science/Data-Visualization/assets/sweep_df.csv
Comment (metaboulie) [inline marimo-tutorials/Data-Science/Data-Visualization/assets/sweep_df.csv:1]: Is this file added by mistake?

**Comment by metaboulie** [comment]
We can collaborate on this if you like

File: marimo-tutorials/artificial-intelligence/deep-learning/activation-functions.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+        {plot_activation(activation_functions.value)[1]}
+        {mo.as_html(plot_activation(activation_functions.value)[0])}
+        """
+    )
+    return np, plot_activation, plt
+
+
+@app.cell(hide_code=True)
+def __(mo):
+    activation_functions = mo.ui.dropdown(
```
Comment (metaboulie) [inline marimo-tutorials/artificial-intelligence/deep-learning/activation-functions.py:118]: I actually think this can be a multiselect.
Then readers can select multiple activation functions at the same time and visualize their differences clearly

File: marimo-tutorials/artificial-intelligence/deep-learning/activation-functions.py [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+
+
+@app.cell(hide_code=True)
+def __(activation_functions, mo):
+    import numpy as np
+    import matplotlib.pyplot as plt
+
+    @mo.cache(pin_modules=True)  # if module versions change later on
+    def plot_activation(func_name):
+        x = np.linspace(-10, 10, 1000)
```
[the remark this answers is not in the document]
Comment (Haleshot) [reply marimo-tutorials/artificial-intelligence/deep-learning/activation-functions.py:49]: That would be a nice addition, will try looking into this.

File: marimo-tutorials/artificial-intelligence/deep-learning/activation-functions.py (hunk 1 above)
Comment (Haleshot) [reply marimo-tutorials/artificial-intelligence/deep-learning/activation-functions.py:64]: _Fair_.

File: marimo-tutorials/Data-Science/Data-Visualization/assets/sweep_df.csv
Comment (Haleshot) [reply marimo-tutorials/Data-Science/Data-Visualization/assets/sweep_df.csv:1]: > Is this file added by mistake?

Ah no; it's part of the pygwalker example I added a while back; not sure why it didn't get added along with that.

File: marimo-tutorials/artificial-intelligence/deep-learning/activation-functions.py (hunk 2 above)
Comment (Haleshot) [reply marimo-tutorials/artificial-intelligence/deep-learning/activation-functions.py:118]: That's a nice suggestion; also, can we also try experimenting with the columns feature (wasim) introduced? I guess multi-select/column approach would be a good addition.

Comment (metaboulie) [reply marimo-tutorials/artificial-intelligence/deep-learning/activation-functions.py:118]: That's still an experimental feature right? Not sure whether it will work well on stable marimo versions

Comment (Haleshot) [reply marimo-tutorials/artificial-intelligence/deep-learning/activation-functions.py:118]: > That's still an experimental feature right? Not sure whether it will work well on stable marimo versions

Ahh right; I always wanted to use that feature ever since it came out; guess I can try it locally to see how it fares.

File: marimo-tutorials/artificial-intelligence/deep-learning/activation-functions.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Haleshot) [reply marimo-tutorials/artificial-intelligence/deep-learning/activation-functions.py]: Resolved using slider values instead of hardcoding.

**Comment by Haleshot** [comment]
Made changes wrt comments given above; most of them should be good to go? Also, tweaked the HeaderWidget to be more consistent in showing "_Show More_" button across categories in the heading of the notebook.

**Review by metaboulie — APPROVED** [review]
Nice job 👍
